fix(postgresql): add primaryKey option for uuid

hairmare opened the pull request:
Adds support for creating a uuid field as primary key directly in a create statement using `uuid('id', { primaryKey: true })` for PostgreSQL dialects.

For now the option is only implemented for PostgreSQL dialects since it should work for all of them and it's only really an issue for CockroachDB.

As far as i can tell, using `uuid('id', { primaryKey: true }).primary()` would still work if you need to support other dialects (the `primary()` would just generate a no-op alter statement).

It's getting late here, so i'll add a docs PR tomorrow. Please let me know if the integration test is too hacky and i'll improve on it.

Fixes #5211

intech commented:
The cockroach dialect needs to be able to set the default value to `gen_random_uuid()`. The full SQL line looks like this:
```sql
UUID PRIMARY KEY DEFAULT gen_random_uuid()
```

OlivierCavadenti commented on test/integration2/schema/misc.spec.js:
```diff
@@ -482,6 +482,47 @@ describe('Schema (misc)', () => {
+                'SELECT c.oid FROM pg_catalog.pg_class c WHERE c.relname = ?',
+                [table_name]
+              )
+              .then((res) => {
```
Use instead await notation

OlivierCavadenti commented on test/integration2/schema/misc.spec.js:
```diff
@@ -482,6 +482,47 @@ describe('Schema (misc)', () => {
+
+                return knex
+                  .raw(
+                    `select c.column_name
```
Can we select and test tolumn type too ?

hairmare commented:
I switched to  using await and implemented the column_type check in 3cb450b8aba794946242f9f32fd759ea4e944e29 and added `default gen_random_uuid()` for the CockroachDB dialect in 9010b3f73022c9156850683af76a71e852f6d5f6.

The tests worked locally, can i have another workflow approval here to see if they still work on GitHub Actions?

hairmare commented:
I also added docs in https://github.com/knex/documentation/pull/419

kibertoad reviewed (changes requested):
TS types also need to be updated

hairmare commented:
@kibertoad thanks for the hint, types have been added in 46b8a699479f7d836268ae46593d4c94f54966ba

I also realised that i had forgotten to register the new CockroachDB unit tests which has now also been fixed.

kibertoad commented:
Thank you! I'll publish new version tonight.
